File: src/courses/Enumeration.java
package courses;


class Enumeration {

enum  Order {
FiRST, SECOND, THIRD
}

enum Order2 {
FiRST("one"),
SECOND("two"),
THIRD("three");

String text;
Order2(String text) {
this.text = text;
}
}

enum  Order3 {
FiRST {
 @Override
 Order3 next() {return  SECOND; }
},
SECOND {
@Override
Order3 next() {return  THIRD; }
},
THIRD  {
@Override
Order3 next() {return  FIRST; }
};

abstract Order3 next();
}

/* enumeration
declare an enumeration
*/
void function1() {
var order = Order.FiRST;
}
void function2() {
Order order = Order.FiRST;
String y = "";
switch (order) {
  case FiRST: y = "one";
    break;
  case SECOND: y = "two";
    break;
  case THIRD: y = "three";
    break;
  default: y = "none";
}

}

void function3() {
Order2 order = Order2.FiRST;
var y = order.text;
}

void function4() {
Order3 order = Order3.FiRST;
var y = order.next();
}
}



}

File: src/courses/Enumeration.kt
package courses


class Enumeration {

enum class Order {
FiRST, SECOND, THIRD
}

enum class Order2(val text: String) {
FiRST("one"),
SECOND("two"),
THIRD("three")
}

enum class Order3 {
FiRST {
    override fun next() = SECOND
},
SECOND {
    override fun next() = THIRD
},
THIRD  {
    override fun next() = FiRST
};
abstract fun next(): Order3
}

/* enumeration
declare an enumeration
*/
fun function1() {
var order = Order.FiRST
}
fun function2() {
var order = Order.FiRST
var y = when(order) {
    Order.FiRST -> "one"
    Order.SECOND -> "two"
    Order.THIRD -> "three"
    else -> "none"
}

}

fun function3() {
var order = Order2.FiRST
var y = order.text

}

fun function4() {
var order = Order3.FiRST
var y = order.next()

}
}



}
